sdk/storage/azure-storage-blob/azure/storage/blob/_download.py

```diff
@@ -718,17 +721,20 @@ def read(self, size: int = -1, *, chars: Optional[int] = None) -> T:
                 self._complete_read()
 
             else:
-                while (chunk := next(chunks_iter, None)) is not None and remaining > 0:
+                while (chunk := next(chunks_iter, None)) is not None and remaining > 0:  # type: ignore [arg-type]
                     chunk_data, content_length = downloader.yield_chunk(chunk)
                     self._download_offset += len(chunk_data)
                     self._raw_download_offset += content_length
-                    self._current_content = self._decoder.decode(
-                        chunk_data, final=self._download_complete) if self._text_mode else chunk_data
```

Comment:
azure\storage\blob\_download.py:727: error: Item "None" of "Optional[IncrementalDecoder]" has no attribute "decode"  [union-attr]